tpu-provisioner/internal/controller/deletion_controller.go

```diff
@@ -101,73 +101,43 @@ func (r *DeletionReconciler) Reconcile(ctx context.Context, req ctrl.Request) (c
 		return ctrl.Result{}, fmt.Errorf("listing nodes in node pool: %w", err)
 	}
 
-	// Ensure no user-Pods are running or pending on the node pool.
-	for _, n := range nodes.Items {
-		var pods corev1.PodList
-		// TODO: Can this be done with a "contains" field match to avoid the outer node loop?
-		if err := r.List(ctx, &pods, client.MatchingFields{".spec.nodeName": n.GetName()}); err != nil {
-			return ctrl.Result{}, fmt.Errorf("listing pods for node: %w", err)
-		}
-
-		for _, p := range pods.Items {
-			// Ignore Pods that have finished.
-			if isDone(&p) {
-				continue
-			}
-
-			// Don't let system Pods prevent downsizing.
-			if p.GetNamespace() == "kube-system" {
-				continue
-			}
-
-			if owner := metav1.GetControllerOf(&p); owner != nil {
-				if owner.Kind == "DaemonSet" || owner.Kind == "Node" {
-					continue
-				}
-			}
-
-			// Must be a user-Pod.
-			lg.V(3).Info("Node in node pool has user pod running, ignoring",
-				"podNode", n.GetName(), "podName", p.GetName(), "podNamespace", p.GetNamespace())
-			return ctrl.Result{}, nil
-		}
-	}
-
-	// If node pool passes deletion check once, reconcile again in a few seconds to give
-	// time for a restarting JobSet to recreate the Jobs and pods that may be assigned to
-	// this node pool.
-	value, exists := r.NodePoolsMarkedForDeletion.Load(nodePoolName)
-	if !exists {
-		lg.Info(fmt.Sprintf("Node pool %q passed deletion check once", nodePoolName))
-		r.NodePoolsMarkedForDeletion.Store(nodePoolName, time.Now())
-		return ctrl.Result{RequeueAfter: r.NodeCriteria.PoolDeletionDelay}, nil
-	}
-
-	// If we haven't reached the node pool deletion check interval, this reconcile was
-	// caused by something else, we can return early, and wait for the manually requeued
-	// reconcile we did after the first deletion check passed.
-	firstDeletionCheckTime := value.(time.Time)
-	if time.Now().Sub(firstDeletionCheckTime) < r.NodeCriteria.PoolDeletionDelay {
+	// If no Node objects are associated with this node pool, it will be cleaned up by the garbage collector.
```

Comment:
This check should never be hit b/c we know that we have a Node object at this point (assuming nothing changed in the last millisecond).

Reply:
Makes sense, removed